packages/calcite-components/src/components/action-menu/action-menu.tsx

```diff
@@ -96,7 +96,7 @@ export class ActionMenu implements LoadableComponent {
 
   @Watch("open")
   openHandler(open: boolean): void {
-    this.activeMenuItemIndex = this.open ? 0 : -1;
+    this.activeMenuItemIndex = this.open ? Math.max(0, this.activeMenuItemIndex) : -1;
```

Comment:
This change is needed because the watcher is turned into a `willUpdate()` after the codemod, and willUpdate runs in batch for all changed properties.

The change is required because before the open watcher would be triggered here, setting `activeMenuItemIndex` to 0:

https://github.com/Esri/calcite-design-system/blob/de835893da0023cfdbeb027c8720be308fda839e/packages/calcite-components/src/components/action-menu/action-menu.tsx#L456

but then the value of activeMenuItemIndex is manually changed here:

https://github.com/Esri/calcite-design-system/blob/de835893da0023cfdbeb027c8720be308fda839e/packages/calcite-components/src/components/action-menu/action-menu.tsx#L463

Because of the `willUpdate()` order change, the activeMenuItemIndex is now set to length-1 and then reset to 0 by the openHandler().

Adding `Math.max(0, this.activeMenuItemIndex)` workarounds this difference.
The other solution is to turn `open` watcher into a getter/setter, but that requires manual changes post-codemod.

Reply:
> This option seems cleaner to me

Makes sense. I will undo
Based on what we discussed, I will add logic to the codemod to convert this specific occurrence of watch into a get/set